core/loadOptions.js

```diff
@@ -76,7 +76,7 @@ var legacyOptionsChecker = function(options, fileName){
 module.exports = function(basePath, _silent){
     silent = _silent;
 
-    var pathToApp = basePath || path.dirname(require.main.filename);
+    var pathToApp = basePath || global.pathToApp;
```

Comment:
Using globals is considered a bad practice, since it requires some special environment setup of the callee. I'll need to double-check how these changes affect the current setup.

From other side `require.main.filename` is also limited, and was the reason to be replaced, since you execute SourceJS code from other location.